src/uwtools/rocoto.py

```diff
@@ -383,6 +384,9 @@ def _set_and_render_jobname(self, config: dict, taskname: str) -> None:
         """
         if STR.jobname not in config:
             config[STR.jobname] = taskname
+        cfg = YAMLConfig(config=config)
+        cfg.dereference()
+        return cfg.data
```

Comment:
Since any of the entries might have values that include a `{{jobname}}` template will need to be updated, it can't be done in place as easily as it can with returning a new dict.